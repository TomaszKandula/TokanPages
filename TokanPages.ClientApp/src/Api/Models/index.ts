export type { AddArticleDto } from "./Articles/addArticleDto";
export type { UpdateArticleContentDto } from "./Articles/updateArticleContentDto";
export type { UpdateArticleCountDto } from "./Articles/updateArticleCountDto";
export type { UpdateArticleLikesDto } from "./Articles/updateArticleLikesDto";
export type { UpdateArticleVisibilityDto } from "./Articles/updateArticleVisibilityDto";
export type { RemoveArticleDto } from "./Articles/removeArticleDto";

export type { SendMessageDto } from "./Mailer/sendMessageDto";
export type { SendNewsletterDto } from "./Mailer/sendNewsletterDto";
export type { SubscriberInfoDto } from "./Mailer/subscriberInfoDto";
export type { VerifyEmailAddressDto } from "./Mailer/verifyEmailAddressDto";

export type { AddSubscriberDto } from "./Subscribers/addSubscriberDto";
export type { UpdateSubscriberDto } from "./Subscribers/updateSubscriberDto";
export type { RemoveSubscriberDto } from "./Subscribers/removeSubscriberDto";

export type { UserDataDto } from "./Users/userDataDto";
export type { ActivateUserDto } from "./Users/activateUserDto";
export type { AddUserDto } from "./Users/addUserDto";
export type { UpdateUserDto } from "./Users/updateUserDto";
export type { RemoveUserDto } from "./Users/removeUserDto";
export type { AuthenticateUserDto } from "./Users/authenticateUserDto";
export type { AuthenticateUserResultDto } from "./Users/authenticateUserResultDto";
export type { ReAuthenticateUserDto } from "./Users/reAuthenticateUserDto";
export type { ReAuthenticateUserResultDto } from "./Users/reAuthenticateUserResultDto";
export type { RevokeUserRefreshTokenDto } from "./Users/revokeUserRefreshTokenDto";
export type { ResetUserPasswordDto } from "./Users/resetUserPasswordDto";
export type { UpdateUserPasswordDto } from "./Users/updateUserPasswordDto";
export type { UserPermissionDto } from "./Users/userPermissionDto";
export type { UserRoleDto } from "./Users/userRoleDto";
export type { UploadUserMediaDto } from "./Users/uploadUserMediaDto";
export type { UploadUserMediaResultDto } from "./Users/uploadUserMediaResultDto";

export type { ErrorDto } from "./Error/errorDto";
export type { ValidationErrorsDto } from "./Error/validationErrorsDto";

export type { DocumentContentDto } from "./Components/documentContentDto";

export type { NavigationContentDto } from "./Components/navigationContentDto";
export type { HeaderContentDto } from "./Components/headerContentDto";
export type { FooterContentDto } from "./Components/footerContentDto";
export type { ClientsContentDto } from "./Components/clientsContentDto";
export type { ArticleFeaturesContentDto } from "./Components/articleFeaturesContentDto";
export type { ContactFormContentDto } from "./Components/contactFormContentDto";
export type { CookiesPromptContentDto } from "./Components/cookiesPromptContentDto";
export type { FeaturedContentDto } from "./Components/featuredContentDto";
export type { FeaturesContentDto } from "./Components/featuresContentDto";
export type { NewsletterContentDto } from "./Components/newsletterContentDto";
export type { ResetPasswordContentDto } from "./Components/resetPasswordContentDto";
export type { UpdatePasswordContentDto } from "./Components/updatePasswordContentDto";
export type { UserSigninContentDto } from "./Components/userSigninContentDto";
export type { UserSignoutContentDto } from "./Components/userSignoutContentDto";
export type { UserSignupContentDto } from "./Components/userSignupContentDto";
export type { TestimonialsContentDto } from "./Components/testimonialsContentDto";
export type { UnsubscribeContentDto } from "./Components/unsubscribeContentDto";
export type { ActivateAccountContentDto } from "./Components/activateAccountContentDto";
export type { UpdateSubscriberContentDto } from "./Components/updateSubscriberContentDto";
export type { WrongPagePromptContentDto } from "./Components/wrongPagePromptContentDto";
export type { 
    AccountContentDto, 
    SectionAccessDenied,
    SectionAccountInformation,
    SectionAccountPassword,
    SectionAccountRemoval
} from "./Components/accountContentDto";

export type { IconDto } from "./Components/Common/iconDto";
export type { LinkDto } from "./Components/Common/linkDto";
export type { RowItemDto } from "./Components/Common/rowItemDto";
export type { TextItemDto } from "./Components/Common/textItemDto";
export type { ContentDto } from "./Components/Common/contentDto";

export type { GetContentManifestDto } from "./Content/getContentManifestDto";
export type { LanguageItemDto } from "./Content/Items/languageItemDto";
export type { JWT } from "./Jwt";
