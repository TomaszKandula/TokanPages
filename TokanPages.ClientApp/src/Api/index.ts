export {
    IsSuccessStatusCode,
    GetBaseHeaders,
    GetProcessedBody,
    GetProcessedHeaders,
    GetProcessedResponse,
} from "./Utilities";

export type { ExecuteApiActionResultProps, ExecuteStoreActionProps, ExecuteApiActionProps } from "./Utilities/Types";

export {
    API_BASE_URI,
    GET_ARTICLES,
    GET_ARTICLE,
    GET_ARTICLE_BY_TITLE,
    ADD_ARTICLE,
    UPDATE_ARTICLE_CONTENT,
    UPDATE_ARTICLE_COUNT,
    UPDATE_ARTICLE_LIKES,
    UPDATE_ARTICLE_VISIBILITY,
    RETRIEVE_ARTICLE_INFO,
    REMOVE_ARTICLE,
    LOG_MESSAGE,
    GET_USERS,
    GET_USER,
    GET_USER_IMAGE,
    ADD_USER,
    ACTIVATE_USER,
    UPDATE_USER,
    REMOVE_USER,
    AUTHENTICATE,
    REAUTHENTICATE,
    VERIFY_USER_EMAIL,
    REVOKE_REFRESH_TOKEN,
    REVOKE_USER_TOKEN,
    RESET_USER_PASSWORD,
    UPDATE_USER_PASSWORD,
    UPLOAD_USER_IMAGE,
    GET_NEWSLETTERS,
    GET_NEWSLETTER,
    ADD_NEWSLETTER,
    UPDATE_NEWSLETTER,
    REMOVE_NEWSLETTER,
    SEND_MESSAGE,
    SEND_NEWSLETTER,
    NOTIFY_WEB_URL,
    NOTIFICATION_STATUS,
    GET_CONTENT_MANIFEST,
    REQUEST_PAGE_DATA,
    GET_ARTICLE_MAIN_IMAGE_URL,
    GET_IMAGES_URL,
    GET_DOCUMENTS_URL,
    ARTICLE_PATH,
    GET_USER_VIDEO,
    GET_VIDEO_ASSET,
    GET_USER_NOTES,
    GET_USER_NOTE,
    ADD_USER_NOTE,
    UPDATE_USER_NOTE,
    REMOVE_USER_NOTE,
} from "./Paths";
