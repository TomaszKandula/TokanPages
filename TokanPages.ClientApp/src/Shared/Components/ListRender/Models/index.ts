export type { Fields } from "./Fields";
export type { Item } from "./Item";
export type { Subitem } from "./Subitem";
