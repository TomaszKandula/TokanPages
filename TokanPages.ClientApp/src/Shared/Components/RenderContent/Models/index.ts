export type { IArticleItem } from "./ArticleItemModel";
export type { IAuthor } from "./AuthorModel";
export type { ITextObject, ITextItem, IRowItem } from "./TextModel";
