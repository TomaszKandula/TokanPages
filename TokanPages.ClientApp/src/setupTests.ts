// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
