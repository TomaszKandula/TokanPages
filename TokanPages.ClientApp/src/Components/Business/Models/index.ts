import { ServiceItemDto, TechItemsDto } from "../../../Api/Models";
import { ReactChangeEvent, ReactMouseEvent } from "../../../Shared/types";

export interface MessageFormProps {
    company: string;
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    description: string;
    techStack: string[];
    services: string[];
}

export interface TechStackListProps {
    isDisabled: boolean;
    list: TechItemsDto[];
    handler: (event: ReactChangeEvent, isChecked: boolean) => void;
}

export interface ServiceItemCardProps {
    isDisabled: boolean;
    key: React.Key | null | undefined;
    value: ServiceItemDto;
    handler: (event: ReactMouseEvent) => void;
    services: string[];
}

export interface BusinessFormProps {
    hasCaption?: boolean;
    hasIcon?: boolean;
    hasShadow?: boolean;
    className?: string;
    background?: string;
}
