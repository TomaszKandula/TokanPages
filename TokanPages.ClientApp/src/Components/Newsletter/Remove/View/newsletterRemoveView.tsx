import * as React from "react";
import { ContentDto } from "../../../../Api/Models";
import { ViewProperties } from "../../../../Shared/Types";
import { CustomCard, Icon, ProgressBar } from "../../../../Shared/Components";
import { ExtendedViewProps } from "../newsletterRemove";

interface NewsletterRemoveViewProps extends ViewProperties, ExtendedViewProps {
    isMobile: boolean;
    hasEmptyId: boolean;
    contentPre: ContentDto;
    contentPost: ContentDto;
    buttonHandler: () => void;
    buttonState: boolean;
    progress: boolean;
    isRemoved: boolean;
}

const ActiveButton = (props: NewsletterRemoveViewProps): React.ReactElement => {
    const content: ContentDto = props.isRemoved ? props.contentPost : props.contentPre;
    return (
        <button
            type="submit"
            onClick={props.buttonHandler}
            className="bulma-button bulma-is-light bulma-is-fullwidth"
            disabled={props.progress || !props.buttonState || props.hasEmptyId}
        >
            {!props.progress ? content.button : <ProgressBar size={20} />}
        </button>
    );
};

export const NewsletterRemoveView = (props: NewsletterRemoveViewProps): React.ReactElement => {
    const content: ContentDto = props.isRemoved ? props.contentPost : props.contentPre;
    return (
        <section className={props.className}>
            <div className="bulma-container bulma-is-max-desktop">
                <div className="py-6">
                    <CustomCard
                        isLoading={props.isLoading}
                        caption={content.caption}
                        text={[content.text1, content.text2, content.text3]}
                        icon={<Icon size={3} name="Email" />}
                        colour="has-text-info"
                        externalButton={<ActiveButton {...props} />}
                    />
                </div>
            </div>
        </section>
    );
};
