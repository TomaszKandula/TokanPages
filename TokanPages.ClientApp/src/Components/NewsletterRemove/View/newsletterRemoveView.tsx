import * as React from "react";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import CircularProgress from "@material-ui/core/CircularProgress";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import { Card, CardContent } from "@material-ui/core";
import Skeleton from "@material-ui/lab/Skeleton";
import { ContentDto } from "../../../Api/Models";
import { ViewProperties } from "../../../Shared/Abstractions";
import { NewsletterRemoveStyle } from "./newsletterRemoveStyle";

interface NewsletterRemoveViewProps extends ViewProperties {
    contentPre: ContentDto;
    contentPost: ContentDto;
    buttonHandler: () => void;
    buttonState: boolean;
    progress: boolean;
    isRemoved: boolean;
    background?: React.CSSProperties;
}

const ActiveButton = (props: NewsletterRemoveViewProps): JSX.Element => {
    const classes = NewsletterRemoveStyle();
    const content: ContentDto = props.isRemoved ? props.contentPost : props.contentPre;
    return (
        <Button
            fullWidth
            type="submit"
            variant="contained"
            onClick={props.buttonHandler}
            className={classes.button}
            disabled={props.progress || !props.buttonState}
        >
            {!props.progress ? content.button : <CircularProgress size={20} />}
        </Button>
    );
};

export const NewsletterRemoveView = (props: NewsletterRemoveViewProps): JSX.Element => {
    const classes = NewsletterRemoveStyle();
    const content: ContentDto = props.isRemoved ? props.contentPost : props.contentPre;
    return (
        <section className={classes.section} style={props.background}>
            <Container maxWidth="sm">
                <Box py={15}>
                    <Card elevation={4}>
                        <CardContent className={classes.card}>
                            <Box textAlign="center" mb={3}>
                                <Box mt={2} mb={2}>
                                    <Typography className={classes.caption}>
                                        {props.isLoading ? <Skeleton variant="text" /> : content.caption}
                                    </Typography>
                                </Box>
                                <Box mt={5} mb={2}>
                                    <Typography className={classes.text1}>
                                        {props.isLoading ? <Skeleton variant="text" /> : content.text1}
                                    </Typography>
                                </Box>
                                <Box mt={5} mb={2}>
                                    <Typography className={classes.text2}>
                                        {props.isLoading ? <Skeleton variant="text" /> : content.text2}
                                    </Typography>
                                </Box>
                                <Box mt={5} mb={7}>
                                    <Typography className={classes.text3}>
                                        {props.isLoading ? <Skeleton variant="text" /> : content.text3}
                                    </Typography>
                                </Box>
                                {props.isLoading ? <Skeleton variant="rect" /> : <ActiveButton {...props} />}
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>
    );
};
