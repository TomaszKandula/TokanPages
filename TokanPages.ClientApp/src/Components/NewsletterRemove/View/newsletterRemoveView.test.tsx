import "../../../setupTests";
import React from "react";
import { shallow } from "enzyme";
import { NewsletterRemoveView } from "./newsletterRemoveView";
import { ContentDto } from "../../../Api/Models";

describe("test component: newsletterRemoveView", () => {
    it("should render correctly '<NewsletterRemoveView />' when content is loaded.", () => {
        const content: ContentDto = {
            caption: "Cancel your subscribtion",
            text1: "We are sorry to see you go...",
            text2: "...but we understand there are circumstances",
            text3: "Please contact us should you have any questions",
            button: "Unsubscribe",
        };

        const tree = shallow(
            <NewsletterRemoveView
                isLoading={false}
                buttonHandler={jest.fn()}
                buttonState={false}
                progress={false}
                contentPre={content}
                contentPost={content}
                isRemoved={false}
            />
        );

        expect(tree).toMatchSnapshot();
    });
});
