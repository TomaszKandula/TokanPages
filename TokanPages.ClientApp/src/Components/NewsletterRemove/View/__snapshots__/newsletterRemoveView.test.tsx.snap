// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component: newsletterRemoveView should render correctly '<NewsletterRemoveView />' when content is loaded. 1`] = `
<section
  class="makeStyles-section-1"
>
  <div
    class="MuiContainer-root makeStyles-container-2 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <div
        class="MuiPaper-root MuiCard-root MuiPaper-elevation4 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-card-7"
        >
          <div
            class="MuiBox-root MuiBox-root-10"
          >
            <div
              class="MuiBox-root MuiBox-root-11"
            >
              <p
                class="MuiTypography-root makeStyles-caption-3 MuiTypography-body1"
              >
                Cancel your subscribtion
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-12"
            >
              <p
                class="MuiTypography-root makeStyles-text1-4 MuiTypography-body1"
              >
                We are sorry to see you go...
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-13"
            >
              <p
                class="MuiTypography-root makeStyles-text2-5 MuiTypography-body1"
              >
                ...but we understand there are circumstances
              </p>
            </div>
            <div
              class="MuiBox-root MuiBox-root-14"
            >
              <p
                class="MuiTypography-root makeStyles-text3-6 MuiTypography-body1"
              >
                Please contact us should you have any questions
              </p>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-22 Mui-disabled MuiButton-fullWidth Mui-disabled"
              disabled=""
              tabindex="-1"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Unsubscribe
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
